### PR TITLE
#3 Create User's and associate points based on random numbers

### PR DESCRIPTION
**Pull request by StoriTella**
#3 Create User model and associate points based on random numbers

### CODE REVIEW

**Review by Kaidey — APPROVED** [review]
Very good very nice

File: src/main/java/ppp/discord/bot/commands/PointsCommand.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import discord4j.core.object.entity.Message;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import ppp.discord.bot.model.User;
+import ppp.discord.bot.utils.LanguageUtil;
+import reactor.core.publisher.Mono;
+
+@Component("pointsCommand")
+public class PointsCommand implements Command<MessageCreateEvent> {
```
Comment (nelinho-rgb) [inline src/main/java/ppp/discord/bot/commands/PointsCommand.java:13]: Try to stick to 1 feature, this commit touches both on the points  and language feature  =)

File: src/main/java/ppp/discord/bot/model/User.java [hunk 2]
```diff
@@ -1,4 +1,67 @@
 package ppp.discord.bot.model;
 
+import discord4j.common.util.Snowflake;
+import discord4j.core.event.domain.message.MessageCreateEvent;
+import java.util.Objects;
+
 public class User {
+
+    private Snowflake id;
```
Comment (nelinho-rgb) [inline src/main/java/ppp/discord/bot/model/User.java:9]: Why are we using this "Snowflake" object?

File: src/main/resources/i18n/i18n_ppt.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "operations": {
+    "success": "Bem me parecia filho da puta",
+    "error": "O que nao maninho nao pode.\nQue estas para ai a dizer levas ja 2 putas",
+    "pong": "ping a tua mae seu fdp",
+    "notSuported": "Mas queres o que mesmo?"
+  },
+  "lang": {
+    "greet": "Come e que e meus putooooooooooooooooooooooos",
```
Comment (nelinho-rgb) [inline src/main/resources/i18n/i18n_ppt.json:9]: Lower the cringe levels =)

File: src/main/java/ppp/discord/bot/commands/PointsCommand.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        return message.getContent().contains(pointKeyGet);
+    }
+
+    public Mono<Void> addPoints(MessageCreateEvent event) {
+        return Mono.empty();
+    }
+
+    public Mono<Void> getPoints(MessageCreateEvent event) {
+        Message eventMessage = event.getMessage();
+        if (getUser(event) != null) {
```
Comment (nelinho-rgb) [inline src/main/java/ppp/discord/bot/commands/PointsCommand.java:74]: Can we try and use Optional? i know it doesn't change much but i kinda like it

File: src/main/java/ppp/discord/bot/utils/LanguageUtil.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+            for (String message: key.split("-")) {
+                if (key.endsWith(message)) {
+                    return (String) jsonObj.get(message);
+                } else {
+                    jsonObj = (JSONObject) jsonObj.get(message);
+                }
+            }
+
+            return "notFound";
```
Comment (nelinho-rgb) [inline src/main/java/ppp/discord/bot/utils/LanguageUtil.java:61]: We could wrap these error messagens around a error object or an enum right? to help with readibility

File: src/main/java/ppp/discord/bot/model/User.java [hunk 6]
```diff
@@ -1,4 +1,67 @@
...
+    }
+
+    public int addPoints(int points) {
+        this.points+=points;
+        return this.points;
+    }
+
+    public int subPoints(int points) {
+        this.points-=points;
+        return this.points;
```
Comment (nelinho-rgb) [inline src/main/java/ppp/discord/bot/model/User.java:46]: we can do this in 1 line right?

File: src/main/java/ppp/discord/bot/commands/PointsCommand.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        Message eventMessage = event.getMessage();
+        this.passivePoints(event);
+        if (isGetPoints(eventMessage)) {
+            return getPoints(event);
+        }
+
+        return Mono.empty();
+    }
+
+    public Mono<Void> passivePoints(MessageCreateEvent event) {
```
Comment (nelinho-rgb) [inline src/main/java/ppp/discord/bot/commands/PointsCommand.java:46]: Try to leave some javadoc for these kind of methods, se we can easily understand what "passive points" are =)

**Review by nelinho-rgb — APPROVED** [review]

File: src/main/java/ppp/discord/bot/model/User.java (hunk 2 above)
Comment (StoriTella) [reply src/main/java/ppp/discord/bot/model/User.java:9]: It's the ID of the Member of discord

**Pull request merged**